src/transformers/generation_utils.py

```diff
@@ -1879,13 +1913,14 @@ def contrastive_search(
                 if this_peer_finished_flag.item() == 0.0:
                     break
 
-            # prepare inputs
-            model_kwargs["use_cache"] = True
-            model_inputs = self.prepare_inputs_for_generation(input_ids, **model_kwargs)
```

Comment:
This was outside the `if` below (that only runs in the 1st iteration), and it is only needed in the first iteration -- moved to inside the `if`